src/Factory/SchemaCacheFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Factory;
+
+use Psr\Container\ContainerInterface;
+use RuntimeException;
+use Yiisoft\Db\Cache\SchemaCache;
+use Yiisoft\Factory\Definitions\DefinitionInterface;
+use Yiisoft\Factory\Definitions\Normalizer;
+use Yiisoft\Factory\Factory;
+
+final class SchemaCacheFactory extends Factory
+{
+    private static ?ContainerInterface $container = null;
+    private static ?DefinitionInterface $definition = null;
+    private static Factory $schemaCacheFactory;
+
+    private function __construct(ContainerInterface $container = null)
+    {
+        self::$container = $container;
+        self::$definition = Normalizer::normalize(SchemaCache::class);
+    }
+
+    public static function initialize(ContainerInterface $container = null): void
+    {
+        self::$schemaCacheFactory = new self($container);
+    }
+
+    /**
+     * Creates a `SchemaCache` instance.
+     *
+     * @throws RuntimeException If the created object is not an instance of the `SchemaCache`.
+     *
+     * @return SchemaCache The `SchemaCache` instance.
+     *
+     * @psalm-suppress RedundantConditionGivenDocblockType
+     * @psalm-suppress DocblockTypeContradiction
+     */
+    public static function run(): SchemaCache
```

Comment:
```suggestion
    public static function create(): SchemaCache
```

Reply:
Can remove inheritance, as it is redundant here. Then you can rename the method from `run()` to `create()`.

Reply:
done.